apps/admin_app/lib/admin_app_web/controllers/property_controller.ex

```diff
@@ -0,0 +1,82 @@
+defmodule AdminAppWeb.PropertyController do
+  use AdminAppWeb, :controller
+
+  alias Snitch.Data.Model.Property, as: PropertyModel
+  alias Snitch.Data.Schema.Property, as: PropertySchema
+
+  def index(conn, params) do
+    properties = PropertyModel.get_all()
+    render(conn, "index.html", %{properties: properties})
+  end
+
+  def new(conn, params) do
+    changeset = PropertySchema.create_changeset(%PropertySchema{}, params)
+    render(conn, "new.html", changeset: changeset)
+  end
+
+  def create(conn, %{"property" => params}) do
+    case PropertyModel.create(params) do
+      {:ok, _} ->
+        redirect(conn, to: property_path(conn, :index))
+
+      {:error, changeset} ->
+        render(conn, "new.html", changeset: %{changeset | action: :new})
+    end
+  end
+
+  def edit(conn, %{"id" => id}) do
+    with {id, _} <- Integer.parse(id),
+         %PropertySchema{} = property <- PropertyModel.get(id) do
+      changeset = PropertySchema.update_changeset(property, %{})
+      render(conn, "edit.html", changeset: changeset)
+    else
+      :error ->
+        conn
+        |> put_flash(:info, "Invalid id #{id}")
+        |> redirect(to: property_path(conn, :index))
+
+      nil ->
+        conn
+        |> put_flash(:info, "No resource found")
+        |> redirect(to: property_path(conn, :index))
+    end
+  end
+
+  def update(conn, %{"id" => id, "property" => params}) do
+    with {id, _} <- Integer.parse(id),
+         property <- PropertyModel.get(id),
+         _ <- PropertyModel.update(property, params) do
+      properties = PropertyModel.get_all()
+      render(conn, "index.html", %{properties: properties})
+    else
+      {:error, changeset} ->
+        render(conn, "edit.html", changeset: %{changeset | action: :edit})
+
+      :error ->
+        conn
+        |> halt()
+        |> put_flash(:info, "Invalid id #{id}")
```

Comment:
Don't think you should show the id here or even mention id. A more general message like property not found would do.